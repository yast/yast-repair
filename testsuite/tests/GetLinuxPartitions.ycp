/*
 * YaST2: Modules testsuite
 *
 * Description:
 *   Testsuite for YaST2 OS Repair: GetLinuxPartitions() in file common_filesystem.ycp
 *
 * Authors:
 *   Michael Koehrmann <curry@suse.de>
 *
 * $Id$
 */
{
    include "testsuite.ycp";
    include "osr/common_filesystem.ycp";

    map target_map = $[];

    // Test 1: ["/dev/hda1", "/dev/hda5", "/dev/hda6", "/dev/sda1", "/dev/sda3"]
    target_map = $[
		   "/dev/hda":$[
				"bios_id":"0x80",
				"bus":"IDE",
				"cyl_count":1583,
				"cyl_size":8224768,
				"model":"ST313021A",
				"name":"1. IDE, 12.13 GB, /dev/hda, ST313021A",
				"partitions":[
					      $["fsid":131, "fstype":"Linux native", "nr":1, "region":[0, 144], "type":`primary],
					      $["fsid":130, "fstype":"Linux swap", "nr":2, "region":[144, 39], "type":`primary],
					      $["fsid":15, "fstype":"Win95 Ext'd (LBA)", "nr":3, "region":[183, 1400], "type":`extended],
					      $["fsid":131, "fstype":"Linux native", "nr":5, "region":[183, 144], "type":`logical],
					      $["fsid":131, "fstype":"Linux native", "nr":6, "region":[327, 173], "type":`logical]
				],
				"scrpath":.hda,
				"vendor":""
		   ],
		   "/dev/sda":$[
				"bios_id":"0x81",
				"bus":"SCSI",
				"cyl_count":699,
				"cyl_size":1048576,
				"model":"LIGHTNING 730S",
				"name":"1. SCSI, 699.00 MB, /dev/sda, QUANTUM-LIGHTNING 730S",
				"partitions":[
					      $["fsid":131, "fstype":"Linux native", "nr":1, "region":[0, 51], "type":`primary],
					      $["fsid":130, "fstype":"Linux swap", "nr":2, "region":[51, 101], "type":`primary],
					      $["fsid":131, "fstype":"Linux native", "nr":3, "region":[152, 547], "type":`primary]],
				"scrpath":.sda,
				"vendor":"QUANTUM"
		   ]
    ];

    TEST(``(GetLinuxPartitions(target_map)), [], nil);

    // Test 2: ["/dev/hda12", "/dev/hda5", "/dev/hda96", "/dev/sda31", "/dev/sda53"]
    target_map = $[
		   "/dev/hda":$[
				"bios_id":"0x80",
				"bus":"IDE",
				"cyl_count":1583,
				"cyl_size":8224768,
				"model":"ST313021A",
				"name":"1. IDE, 12.13 GB, /dev/hda, ST313021A",
				"partitions":[
					      $["fsid":131, "fstype":"Linux native", "nr":12, "region":[0, 144], "type":`primary],
					      $["fsid":130, "fstype":"Linux swap", "nr":23, "region":[144, 39], "type":`primary],
					      $["fsid":15, "fstype":"Win95 Ext'd (LBA)", "nr":35, "region":[183, 1400], "type":`extended],
					      $["fsid":131, "fstype":"Linux native", "nr":5, "region":[183, 144], "type":`logical],
					      $["fsid":131, "fstype":"Linux native", "nr":96, "region":[327, 173], "type":`logical]
				],
				"scrpath":.hda,
				"vendor":""
		   ],
		   "/dev/sda":$[
				"bios_id":"0x81",
				"bus":"SCSI",
				"cyl_count":699,
				"cyl_size":1048576,
				"model":"LIGHTNING 730S",
				"name":"1. SCSI, 699.00 MB, /dev/sda, QUANTUM-LIGHTNING 730S",
				"partitions":[
					      $["fsid":131, "fstype":"Linux native", "nr":31, "region":[0, 51], "type":`primary],
					      $["fsid":130, "fstype":"Linux swap", "nr":24, "region":[51, 101], "type":`primary],
					      $["fsid":131, "fstype":"Linux native", "nr":53, "region":[152, 547], "type":`primary]],
				"scrpath":.sda,
				"vendor":"QUANTUM"
		   ]
    ];

    TEST(``(GetLinuxPartitions(target_map)), [], nil);
    
}
