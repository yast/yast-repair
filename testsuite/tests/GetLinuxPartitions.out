Return	["/dev/hda1", "/dev/hda5", "/dev/hda6", "/dev/sda1", "/dev/sda3"]
Return	["/dev/hda12", "/dev/hda5", "/dev/hda96", "/dev/sda31", "/dev/sda53"]
